src/with-defaults.ts

```diff
@@ -4,30 +4,30 @@ import endpointWithDefaults = require('./endpoint-with-defaults')
 import merge = require('./merge')
 import parse = require('./parse')
 
-interface withDefaults {
-  (route: any, options: any): any
-  DEFAULTS: (typeof import('./defaults') | null) & any
-  defaults: (newDefaults: {
-    method: string;
-    baseUrl: string;
-    headers: {
-      accept: string;
-      'user-agent': string;
-    };
-    mediaType: {
-      format: string;
-      previews: string[];
-    };
-  }) => withDefaults;
-  merge: (route: any, options?: any) => typeof import('./defaults')
-}
+import { EndpointOptions, EndpointDefaultOptions, endpoint, Route, RouteOptions } from './types'
+
+function withDefaults(oldDefaults: EndpointDefaultOptions | null, newDefaults: RouteOptions): endpoint {
+  const DEFAULTS = merge(oldDefaults, newDefaults)
+  // why can’t we just do this: ¯\_(ツ)_/¯
+  // const endpoint = endpointWithDefaults.bind(null, DEFAULTS)
+  const endpoint = (route: Route | EndpointOptions, options?: RouteOptions) => {
+    if (typeof route === 'string') {
+      return endpointWithDefaults(DEFAULTS, route, options)
+    }
 
-function withDefaults(oldDefaults: typeof import('./defaults') | null, newDefaults: typeof import('./defaults')): withDefaults {
-  const DEFAULTS: typeof oldDefaults & typeof newDefaults = merge(oldDefaults, newDefaults)
-  return Object.assign(endpointWithDefaults.bind(null, DEFAULTS), {
+    return endpointWithDefaults(DEFAULTS, route)
+  }
+
+  return Object.assign(endpoint, {
     DEFAULTS,
     defaults: withDefaults.bind(null, DEFAULTS),
-    merge: merge.bind(null, DEFAULTS),
+    merge: (route: Route | RouteOptions, options?: RouteOptions) => {
+      if (typeof route === 'string') {
+        return merge(DEFAULTS, route, options)
+      }
+
+      return merge(DEFAULTS, route)
+    },
```

Comment:
same as above. This `typeof route === 'string'` just blows up the source code without doing anything useful :(